sdk/data_cosmos/src/cosmos_entity.rs

```diff
@@ -44,11 +44,20 @@ impl CosmosEntity for serde_json::Value {
 /// Serialize the partition key in the format CosmosDB expects.
 pub(crate) fn serialize_partition_key<PK: Serialize>(pk: &PK) -> azure_core::Result<String> {
     use azure_core::error::ResultExt;
-    // this must be serialized as an array even tough CosmosDB supports only a single partition key.
-    serde_json::to_string(&[pk]).context(
+
+    // if pk is array, we dont need to wrap it in an array
+    // if it is not, we need to wrap it in an array
+
+    let serialized_string = serde_json::to_string(pk).context(
         azure_core::error::ErrorKind::DataConversion,
         "could not convert partition_key into String",
-    )
+    )?;
+
+    if serialized_string.starts_with('[') {
```

Comment:
Although this will work, can this be implemented without checking the serialization output?

Reply:
I tried changing some of the existing tests, although they relied on predefined `result` data that was asserted against so I left it be. Perhaps I can write some in-file tests?

The only way I could imagine achieving this without checking the serialized text is matching on the implementation of the generic type or something along those lines. This is method is more coupled, although safer as it works for all vector-like types or anything that serializes down to an array in JSON.